src/compiler/checker.ts

```diff
@@ -7809,7 +7812,8 @@ namespace ts {
                     }
                 }
             }
-            if (!props) {
+            const props = arrayFrom(propSet.values());

```

Comment:
Can we guarantee order of properties created from map? We seem to rely on the order eg to set `firstValueDeclaration`

Reply:
Yes. More reliably then objects, even. From [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map):

> The keys in Map are ordered while keys added to object are not. Thus, when iterating over it, a Map object returns keys in order of insertion.